Use public sync API in production code

donat opened the pull request:
The last missing piece for #752. The PR changes Buildship such that all production code executes project synchronization via the public `GradleWorkspace` service.

donat commented:
@marcphilipp It seems like I can't request a review from you if you are not part of the `eclipse` organization. Let's see if it changes after I mention you :)

donat commented:
It does not work. Let's figure something together tomorrow.

marcphilipp commented on org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/internal/wizard/project/ProjectImportConfiguration.java:
```diff
@@ -62,7 +62,7 @@ public void setProjectDir(File projectDir) {
 
 
-    public Property<Boolean> getOverwriteWorkspaceSettings() {
+    public Property<Boolean> overrideWorkspaceConfiguration() {
```
I think this should be called `getOverrideWorkspaceConfiguration()`

donat replied:
Fixed.

donat commented:
I was thinking about removing the interfaces but decided to keep it simple. But since you are also noticed it, I've removed the `GradleBuilds` and replaced the only usage with `Set<GradleBuild>`.
